client/src/featureform/serving.py

```diff
@@ -387,10 +387,41 @@ def get_input_df(self, source_name, source_variant):
             else:
                 raise ValueError(f"Unsupported file format for {file_path}")
             return df
+        elif (
+            self.db.is_transformation(source_name, source_variant)
+            == SourceType.DIRECTORY
+        ):
+            source = self.db.get_source_variant(source_name, source_variant)
+            directory = source["definition"]
+            return self.read_directory(directory)
+
         else:
             df = self.process_transformation(source_name, source_variant)
         return df
 
+    def read_directory(self, directory):
+        if not os.path.isdir(directory):
+            raise Exception(f"Path {directory} is not a directory")
+
+        def absolute_file_paths(directory):
+            for dirpath, _, filenames in os.walk(directory):
+                for f in filenames:
+                    yield os.path.abspath(os.path.join(dirpath, f)), f
+
+        file_names = []
+        file_body = []
+        for absolute_fn, relative_fn in absolute_file_paths(directory):
+            file_names.append(relative_fn)
+            f = open(absolute_fn, "r")
+            try:
+                file_body.append(f.read())
+            except Exception as e:
+                raise Exception(
+                    f"Cannot read file {absolute_fn}: {e}\nFiles must be text files"
```

Comment:
you're capturing all generic exceptions into one here. Do we want to fail the whole thing if one of the files is corrupt/not a text file or do we want to skip over it. 

Also, feels like this check should be done during the register.

Could also do it like this:

```
    try:
        file_data = [
            (relative_fn, Path(absolute_fn).read_text())
            for absolute_fn, relative_fn in absolute_file_paths(directory)
        ]
    except Exception as e:
        raise IOError(f"Cannot read file {absolute_fn}: {e}")

    file_names, file_bodies = zip(*file_data)
    df = pd.DataFrame(data={"filename": file_names, "body": file_bodies})

    return df
```

if you use read_text you don't have to open it but you have to wrap it in Path, it takes care of it for you.